crates/pessimistic-proof/src/local_state.rs

```diff
@@ -248,6 +208,30 @@ impl LocalNetworkState {
                 .verify_and_update(*token, balance_path, *old_balance, new_balance)?;
         }
 
+        // Verify that the signature is valid
+        let combined_hash = signature_commitment(
+            self.exit_tree.get_root(),
+            multi_batch_header
+                .imported_bridge_exits
+                .clone()
+                .into_iter()
+                .map(|(exit, _)| exit)
+                .collect::<Vec<_>>()
+                .as_slice(),
```

Comment:
This depends on the other change suggesting generalization to `signature_commitment`.

In general, avoiding cloning and temporary data structures should save a bunch of cycles.

```suggestion
            multi_batch_header
                .imported_bridge_exits
                .iter()
                .map(|(exit, _)| exit),
```
(this may be missing some trivial refs / derefs)